pkg/entities/tokens.go

```diff
@@ -2,85 +2,73 @@ package entities
 
 import (
 	"encoding/json"
-	"io/ioutil"
+	"errors"
 	"net/http"
 	"strings"
 
 	"github.com/defipod/mochi/pkg/model"
 	"github.com/defipod/mochi/pkg/request"
+	"github.com/defipod/mochi/pkg/response"
+	"gorm.io/gorm"
 )
 
-type Response struct {
-	Id     string `json:"id"`
-	Symbol string `json:"symbol"`
-	Name   string `json:"name"`
-}
-
-func (e *Entity) GetIDAndName(symbol string) (string, string, error) {
-	resp, err := http.Get("https://api.coingecko.com/api/v3/coins/list")
-	// Get request
+func (e *Entity) CreateCustomToken(req request.UpsertCustomTokenConfigRequest) error {
+	token, err := e.repo.Token.GetOneBySymbol(req.Symbol)
 	if err != nil {
-		return "", "", err
+		e.log.Error(err, "[Entity][CreateCustomToken] repo.Token.GetOneBySymbol failed")
 	}
-	defer resp.Body.Close()
 
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return "", "", err
-	}
-	var res []Response
-	err1 := json.Unmarshal([]byte(body), &res)
-	if err1 != nil {
-		return "", "", err
-	}
-	for i := 0; i < len(res); i++ {
-		if res[i].Symbol == symbol {
-			return res[i].Id, res[i].Name, nil
+	if token == nil {
+		chain, err := e.repo.Chain.GetByCurrency(req.Chain)
+		if err != nil {
+			return err
 		}
-	}
 
-	return "", "", err
-}
+		coinGeko, err := getCoinGeckoBySymbol(req.Symbol)
+		if err != nil {
+			return err
+		}
+		if coinGeko == nil {
+			return errors.New("cannot find token by symbol")
+		}
 
-func (e *Entity) CheckExistToken(symbol string) (bool, error) {
-	listSymbol, err := e.repo.Token.GetAll()
-	if err != nil {
-		return false, err
+		token = &model.Token{
+			ChainID:             chain.ID,
+			Address:             req.Address,
+			CoinGeckoID:         coinGeko.ID,
+			Name:                coinGeko.Name,
+			Symbol:              strings.ToUpper(coinGeko.Symbol),
+			Decimals:            req.Decimals,
+			DiscordBotSupported: true,
+			Chain:               chain,
+		}
+		if err = e.repo.Token.CreateOne(token); err != nil {
+			return err
+		}
 	}
 
-	for i, _ := range listSymbol {
-		if strings.ToLower(symbol) == strings.ToLower(listSymbol[i].Symbol) {
-			return true, nil
-		}
+	if !token.DiscordBotSupported {
+		return errors.New("token is not supported to add to your server")
 	}
 
-	return false, nil
-}
+	if !strings.EqualFold(token.Chain.Currency, req.Chain) {
```

Comment:
why compare chain currency with chain?

Reply:
because req.Chain is currency value of chain